### PR TITLE
Sphinx doc chapter 29

### PR DESCRIPTION
**Pull request by maximedenes**
@mattam82 Could you help backporting the changes done on universe documentation since coq/coq@03a71a241f8d05f6a86f3c8f3c2146c4db378f7b ? Thanks!

### CODE REVIEW

**Comment by Zimmi48** [comment]
Shouldn't I wait for the PR to be complete before reviewing?

**Comment by maximedenes** [comment]
> Shouldn't I wait for the PR to be complete before reviewing?

Yes, sorry I wasn't clear.

**Comment by Zimmi48** [comment]
Who's going to take care of this?

I can already foresee a sub-section in my thesis called "The nightmare of the Sphinx migration or how things can still go terribly wrong!"

**Comment by maximedenes** [comment]
@mattam82 any update on this? It is getting critical for the release.

**Comment by mattam82** [comment]
Will do this afternoon

**Comment by Zimmi48** [comment]
I think it would indeed make sense to squash this commit with the previous one, and in this case it would also make reviewing easier.
Can you rebase on master at the same time?

**Comment by mattam82** [comment]
Ok, done!

File: doc/sphinx/addendum/universe-polymorphism.rst [hunk 1]
```diff
@@ -1,393 +1,444 @@
...
+-------------------------
+
+.. cmd:: Polymorphic @definition
+
+   As shown in the examples, polymorphic definitions and inductives can be
+   declared using the ``Polymorphic`` prefix.
+
+.. opt:: Universe Polymorphism
+
+   Once enabled, this option will implicitly prepend ``Polymorphic`` it to any
```
Comment (Zimmi48) [inline doc/sphinx/addendum/universe-polymorphism.rst]: This `it` is a left-over from the old formulation.

File: doc/sphinx/addendum/universe-polymorphism.rst [hunk 2]
```diff
@@ -1,393 +1,444 @@
...
+prefix.
+
+.. cmd:: NonCumulative @inductive
+
+   Declares the inductive as non-cumulative
+
+.. opt:: Polymorphic Inductive Cumulativity
+
+   Sets all following polymorphic inductives types as cumulative or not
+   by default.
```
Comment (Zimmi48) [inline doc/sphinx/addendum/universe-polymorphism.rst]: I would rather say "When this option is on, it sets all following polymorphic inductive types as cumulative (it is off by default)."

File: doc/sphinx/addendum/universe-polymorphism.rst [hunk 3]
```diff
@@ -1,393 +1,444 @@
...
+This definition is elaborated by minimizing the universe of :g:`id0` to
+level :g:`Set` while the more general definition would keep the fresh level
+:g:`i` generated at the application of :g:`id` and a constraint that :g:`Set` :math:`≤ i`.
+This minimization process is applied only to fresh universe variables.
+It simply adds an equation between the variable and its lower bound if
+it is an atomic universe (i.e. not an algebraic max() universe).
+
+.. opt:: Universe Minimization ToSet
+
+   Unsetting this option disallows minimization to the sort :g:`Set` and only
```
Comment (Zimmi48) [inline doc/sphinx/addendum/universe-polymorphism.rst]: I would say "Turning this option off (it is on by default) disallows..."

File: doc/sphinx/addendum/universe-polymorphism.rst [hunk 4]
```diff
@@ -1,393 +1,444 @@
...
+
+   Print le.
+
+During refinement we find that :g:`j` must be larger or equal than :g:`i`, as we
+are using :g:`A : Type@{i} <= Type@{j}`, hence the generated constraint. At the
+end of a definition or proof, we check that the only remaining
+universes are the ones declared. In the term and in general in proof
+mode, introduced universe names can be referred to in terms. Note that
+local universe names shadow global universe names. During a proof, one
+can use Show Universes to display the current context of universes.
```
Comment (Zimmi48) [inline doc/sphinx/addendum/universe-polymorphism.rst]: Missing verbatim. And is this command indexed? (Does it have a `.. cmd::`?)

**Comment by Zimmi48** [comment]
All of my remarks are of minor importance. @mattam82 Can you tell if you plan to address them or would rather have this PR merged without delay?

**Comment by mattam82** [comment]
I’ll fix them now
Le jeu. 12 avr. 2018 à 09:58, Théo Zimmermann <notifications@github.com> a
écrit :

File: doc/sphinx/addendum/universe-polymorphism.rst (hunk 4 above)
Comment (mattam82) [reply doc/sphinx/addendum/universe-polymorphism.rst]: It was wrongly indexed as an exn

File: doc/sphinx/addendum/universe-polymorphism.rst (hunk 3 above)
Comment (mattam82) [reply doc/sphinx/addendum/universe-polymorphism.rst]: Yup

File: doc/sphinx/addendum/universe-polymorphism.rst (hunk 2 above)
Comment (mattam82) [reply doc/sphinx/addendum/universe-polymorphism.rst]: Yup

File: doc/sphinx/addendum/universe-polymorphism.rst (hunk 1 above)
Comment (mattam82) [reply doc/sphinx/addendum/universe-polymorphism.rst]: Fixed

**Comment by mattam82** [comment]
I fixed all your comments @Zimmi48 . Should be ready for merge now.

**Pull request merged**